src/azul/fedramp_report_service.py

```diff
@@ -0,0 +1,494 @@
+
+    def _supported_resource_types(self) -> set[str]:
+        return {'AWS::S3::Bucket'}
+
+    def map(self, resource: JSON) -> Iterator[InventoryRow]:
+        yield InventoryRow(
+            asset_tag=self._get_asset_tag(resource),
+            asset_type='AWS S3 Bucket',
+            comments=self._get_encryption_status(resource),
+            is_public='Yes' if self._get_is_public(resource) else 'No',
+            is_virtual='Yes',
+            location=resource['awsRegion'],
+            software_vendor='AWS',
+            system_owner=self._get_owner(resource),
+            unique_id=resource['arn'],
+        )
+
+    def _get_is_public(self, resource: JSON) -> bool:
+        try:
+            public_access_config = resource['supplementaryConfiguration']['PublicAccessBlockConfiguration']
+        except KeyError:
+            # If there is no PublicAccessBlockConfiguration then this bucket is public
+            return True
+        else:
+            # The bucket is public if any access blocks are false
+            return not all(public_access_config.values())
+
+    def _get_encryption_status(self, resource: JSON) -> str:
+        if 'ServerSideEncryptionConfiguration' in resource['supplementaryConfiguration']:
+            return 'Encrypted'
+        else:
+            return 'Not encrypted'
+
+
+class VPCMapper(Mapper):
+
+    def _supported_resource_types(self) -> set[str]:
+        return {'AWS::EC2::VPC'}
+
+    def map(self, resource: JSON) -> Iterator[InventoryRow]:
+        yield InventoryRow(
+            asset_tag=self._get_asset_tag(resource),
+            asset_type='AWS VPC',
+            baseline_config=resource['configurationStateId'],
+            ip_address=resource['configuration']['cidrBlock'],
+            is_public='Yes',
+            is_virtual='Yes',
+            location=resource['awsRegion'],
+            network_id=resource['configuration']['vpcId'],
+            software_vendor='AWS',
+            system_owner=self._get_owner(resource),
+            unique_id=resource['arn'],
+        )
+
+
+class LambdaMapper(Mapper):
+
+    def _supported_resource_types(self) -> set[str]:
+        return {'AWS::Lambda::Function'}
+
+    def map(self, resource: JSON) -> Iterator[InventoryRow]:
+        configuration = resource['configuration']
+        yield InventoryRow(
+            asset_tag=self._get_asset_tag(resource),
+            asset_type='AWS Lambda Function',
+            baseline_config=configuration['runtime'],
+            is_public='No',
+            is_virtual='Yes',
+            location=resource['awsRegion'],
+            purpose=configuration.get('description'),
+            software_product_name='AWS Lambda',
+            software_vendor='AWS',
+            system_owner=self._get_owner(resource),
+            unique_id=resource['arn'],
+        )
+
+
+class ElasticSearchMapper(Mapper):
+
+    def _supported_resource_types(self) -> set[str]:
+        return {'AWS::Elasticsearch::Domain'}
```

Comment:
Interesting, Amazon didn't rename this?

Reply:
https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html#amazonopensearchservice

Reply:
Listing the resources in `us-east-1` shows 2 instances of `AWS::Elasticsearch::Domain` and none of `AWS::OpenSearch::Domain`:

```
2023-03-06 15:55:19,813 DEBUG   MainThread: Mapping 'AWS::Elasticsearch::Domain' resource using 'ElasticSearchMapper'
2023-03-06 15:55:19,813 DEBUG   MainThread: Mapped to 1 rows
2023-03-06 15:55:19,813 DEBUG   MainThread: Mapping 'AWS::Elasticsearch::Domain' resource using 'ElasticSearchMapper'
2023-03-06 15:55:19,814 DEBUG   MainThread: Mapped to 1 rows
```